pkgs/by-name/la/lazyjj/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  binutils-unwrapped,
+  fetchFromGitHub,
+  makeWrapper,
+  jujutsu,
+  rustPlatform,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "lazyjj";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "Cretezy";
+    repo = "lazyjj";
+    rev = "v${version}";
+    hash = "sha256-VlGmOdF/XsrZ/9vQ14UuK96LIK8NIkPZk4G4mbS8brg=";
+  };
+
+  cargoHash = "sha256-TAq9FufGsNVsmqCE41REltYRSSLihWJwTMoj0bTxdFc=";
+
+  postInstall = ''
+    # mv $out/bin/lazyjj $out/bin/lazyjj-unwrapped
+    wrapProgram $out/bin/lazyjj \
+      --prefix PATH : ${
+        lib.makeBinPath [
+          binutils-unwrapped
+          jujutsu
+        ]
+      }
+  '';
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  nativeCheckInputs = [
+    binutils-unwrapped
+    jujutsu
+  ];
+
+  meta = with lib; {
+    description = "TUI for Jujutsu/jj";
+    homepage = "https://github.com/Cretezy/lazyjj";
+    license = licenses.asl20;
+    mainProgram = "lazyjj";
```

Comment:
suggestion: `meta.mainProgram` should be redundant if it's the same as `pname`.

```suggestion
```

Reply:
No, every package should have `mainProgram` (where applicable). We specifically did a tree‐wide sweep to add it to get rid of the `pname` heuristic, which is explicitly deprecated. Please add this back.

https://github.com/NixOS/nixpkgs/blob/8c2422921c683254fd46053e9fe449c090b15bc9/lib/meta.nix#L398-L403

Reply:
added back, thanks for the explainer!